host/examples/rx_ascii_art_dft.cpp

```diff
@@ -169,20 +170,115 @@ int UHD_SAFE_MAIN(int argc, char *argv[]){
             ascii_art_dft::log_pwr_dft(&buff.front(), num_rx_samps)
         );
         std::string frame = ascii_art_dft::dft_to_plot(
-            lpdft, COLS, LINES,
+            lpdft, COLS, LINES-5,
             usrp->get_rx_rate(),
             usrp->get_rx_freq(),
             dyn_rng, ref_lvl
         );
+        
+        std::string border = std::string((COLS), '-');
 
         //curses screen handling: clear and print frame
         clear();
+        printw("%s", border.c_str());
+        printw("[f-F]req: %4.3f MHz   |   [r-R]ate: %2.2f Msps   |   [b-B]w: %2.2f MHz   |   [g-G]ain: %2.0f dB\n\n", freq/1e6, rate/1e6, bw/1e6, gain);
+        printw("[d-D]yn Range: %2.0f dB    |   Ref [l-L]evel: %2.0f dB   |   fp[s-S] : %2.0f   |   [t-T]uning step: %3.3f M\n", dyn_rng, ref_lvl, frame_rate, step/1e6);
+        printw("%s", border.c_str());       
         printw("%s", frame.c_str());
 
         //curses key handling: no timeout, any key to exit
         timeout(0);
         int ch = getch();
-        if (ch != KEY_RESIZE and ch != ERR) break;
+        
+        // Key handling. FIXME: Do proper bounds check on all parameters
+        if (ch == 'r')
+        {
+            if ((rate - step) > 0) 
+            {
+                rate -= step;                
+                usrp->set_rx_rate(rate);
+            }           
+        }
+
+        else if (ch == 'R')
+        {
+                rate += step;
+                usrp->set_rx_rate(rate);        
+        }
+
+        else if (ch == 'g')
+        {
+            if (gain > 1)   
+            {
+                gain -= 1;
+                usrp->set_rx_gain(gain);
+            }
+        }
+
+        else if (ch == 'G')
+        {
+                gain += 1;
+                usrp->set_rx_gain(gain);
+            
+        }
+
+        else if (ch == 'b')
+        {
+                bw -= step;
+                usrp->set_rx_bandwidth(bw);            
+        }
+
+        else if (ch == 'B')
+        {
+                bw += step;
+                usrp->set_rx_bandwidth(bw);            
+        }
+
+        else if (ch == 'f')
+        {
+            if ((freq - step) < 0)
```

Comment:
A) I think this should be `((freq - step) > 0)`, right?
B) Avoid this by letting the setter handle the bounds checking.
```suggestion
            freq -= step;
            usrp->set_rx_freq(freq);
            freq = usrp->get_rx_freq();
```